addons/account/tests/test_multivat.py

```diff
@@ -0,0 +1,159 @@
+            Setups a company with a custom chart template, containing a tax and a fiscal position.
+            We need to add xml_ids to the templates because they are loaded from their xml_ids
+        """
+        super().setUpClass()
+
+        # Create user.
+        user = cls.env['res.users'].create({
+            'name': 'Because I am accountman!',
+            'login': 'accountman',
+            'password': 'accountman',
+            'groups_id': [Command.set(cls.env.user.groups_id.ids), Command.link(cls.env.ref('account.group_account_user').id)],
+        })
+        user.partner_id.email = 'accountman@test.com'
+
+        cls.company_1 = cls.env['res.company'].create({
+            'name': 'TestCompany1',
+            'country_id': cls.env.ref('base.be').id,
+        })
+
+        cls.env = cls.env(user=user)
+        cls.cr = cls.env.cr
+
+        user.write({
+            'company_ids': [Command.set(cls.company_1.ids)],
+            'company_id': cls.company_1.id,
+        })
+
+        test_get_data = data_method_provider("local", "be")
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=test_get_data, autospec=True):
+            cls.env['account.chart.template'].try_loading('local', company=cls.company_1, install_demo=False)
+
+    def test_create_foreign_tax(self):
+        foreign_country = self.env.ref("base.fr")
+        self.foreign_vat_fpos = self.env["account.fiscal.position"].create({
+            "name": "FR foreign VAT",
+            "auto_apply": True,
+            "country_id": foreign_country.id,
+            "foreign_vat": "FR23334175221",
+        })
+
+        test_get_data = data_method_provider("foreign", "fr")
+        with patch.object(AccountChartTemplate, '_get_chart_template_data', side_effect=test_get_data, autospec=True):
+            self.foreign_vat_fpos.action_create_foreign_taxes()
+
+        # ensure the local tax external_id follows that pattern and is reachable using both api
+        self.env.ref(f'account.{self.env.company.id}_test_tax_1_template')
+        self.env["account.chart.template"].ref('test_tax_1_template')
+        self.env.ref(f'account.{self.env.company.id}_test_tax_2_template')
+        self.env["account.chart.template"].ref('test_tax_2_template')
+
+        # ensure the foreign tax external_id follows that pattern and is reachable using both api
+        self.env.ref(f'account.{self.env.company.id}_foreign_test_tax_1_template')
+        self.env["account.chart.template"].ref('foreign_test_tax_1_template')
+        self.env.ref(f'account.{self.env.company.id}_foreign_test_tax_2_template')
+        tax = self.env["account.chart.template"].ref('foreign_test_tax_2_template')
+
+        self.assertTrue(tax, "a foreign tax should have been instantiated")
+        base_line, tax_line = tax.invoice_repartition_line_ids
+        self.assertNotEqual(base_line.account_id, tax_line.account_id, "The tax line should have been set with another account")
```

Comment:
The problem is that the account is created on the fly without any xml_id.
Thus the only way to do it would be a search and it could fail because it's possible that several accounts with kind of the same name gets [created on the fly](https://github.com/odoo/odoo/blob/d34151f4693a370c6194592c5dea4c4a50224fe6/addons/account/models/chart_template.py#LL651C57-L651C57).